src/Section/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  DetailedHTMLProps,
+  HTMLAttributes,
+  PropsWithChildren,
+} from 'react'
+import { useId } from 'react-id-generator'
+import styled from 'styled-components'
+
+import { HeadingLevel, HX } from '../Heading'
+
+export interface SectionProps {
+  headingLevel: HeadingLevel
+  headingSize?: 'string'
+  headingGap?: 'string'
+
+  title: string
+
+  background?: string
+  padding?: string
+  width?: string
+  height?: string
+}
+
+interface SectionWrapperProps
+  extends DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement> {
+  background: string
+  padding: string
+  width: string
+  height: string
+}
+
+interface SectionHeaderProps {
+  size: string
+  gap: string
+}
+
+const Section: React.FC<PropsWithChildren<SectionProps>> = ({
+  title,
+  children,
+  headingLevel,
+  headingGap = '0.5em',
+  headingSize = '1.2em',
+  background = '#f0f0f0',
+  padding = '1em',
+  width = '100ch',
+  height = 'auto',
+}) => {
+  const [labelId] = useId()
+
+  return (
+    <Wrapper
+      aria-labelledby={labelId}
+      background={background}
+      padding={padding}
+      width={width}
+      height={height}
+    >
+      <Heading headingLevel={headingLevel} size={headingSize} gap={headingGap}>
+        {title}
+      </Heading>
+      {children}
+    </Wrapper>
+  )
+}
+
+const Wrapper = styled.section<SectionWrapperProps>`
+  background: ${({ background }) => background};
```

Comment:
looks like this almost could be done by using spread operator, did not check alll props

Reply:
like `{ headingLevel, headingGap, children, ...rest }` but it might introduce an extra complexity or make it harder to understand.

Reply:
yes maybe, I think I can see it both ways tbh